### PR TITLE
Fix/create an independent program quits

### PR DESCRIPTION
**Pull request by ujjwalpathaak**
## What this PR does
#5532 - If you try to set the time before picking a date, the process will stop and take you back to the main screen. This happens because, without a date, the function can't check the time format properly.

To make things smoother for users, the time fields will be inactive until you choose a date. This way, you won't be able to set the time until you've selected a date first. Once you pick a date, then you can set the time. This should make the whole process more intuitive.

## Screenshots
Before:
image

After:
image

### CODE REVIEW

**Comment by ragesoss** [comment]
Looks like there are some linting errors.

I tested it out manually, and it looks like this will work.

**Comment by ujjwalpathaak** [comment]
I have resolved the linting issues which where mentioned.
image

**Comment by ujjwalpathaak** [comment]
@ragesoss I have made the changes.

**Comment by ragesoss** [comment]
It looks like there might be a behavior change that is affecting one or two the tests. Can you check whether the two feature tests that failed in CI also fail locally for you? If so, you may need to adjust the tests to work with the new datetime behavior.

**Comment by ujjwalpathaak** [comment]
Hello, @ragesoss I encountered several issues with my previous approach due to how date input was being handled.

I've taken a different approach. Now, there's a reducer call to notification.js, and an error message is prompted, saying to enter the date before the time.

While my yarn test cases are passing successfully, I'm feeling a bit uncertain about the rspec test cases. I am new to unit testing and Ruby.

According to me, the test cases affected by my changes are passing, but it would be great if I could take a look at the CI to confirm whether any further adjustments are needed.

image

**Comment by ujjwalpathaak** [comment]
Hello @ragesoss all the checks have passed for the last commit. Can you please have a look at my approach?

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -41,7 +41,11 @@ def fingerprinted(path, filename, file_prefix = nil)
   def css_fingerprinted(filename)
     manifest_path = "#{Rails.root}/public/assets/javascripts/manifest.json"
     manifest = Oj.load(File.read(File.expand_path(manifest_path, __FILE__)))
-    manifest[filename].split('/').last
+    if manifest[filename]
```
Comment (ragesoss) [inline app/helpers/application_helper.rb:44]: This seems unrelated. What is this change for?

**Comment by ragesoss** [comment]
I think the error notice is a little too aggressive. A better option might be to automatically set the date to today if the user clicks the time picker before the date is set.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (ujjwalpathaak) [reply app/helpers/application_helper.rb:44]: I'll stash that change, it was for one of the test failing locally for me but i guess it’s unnecessary and was probably a local issue

**Comment by ujjwalpathaak** [comment]
@ragesoss Hey i guess i made a mistake. Can you please have a look and please resolve this?

**Comment by ragesoss** [comment]
Can you explain a little more? What should I have a look at?

**Comment by ujjwalpathaak** [comment]
I was trying to make some changes in my fork but I ended up force merging this pull request and now it shows it's merged and closed.

image
image

**Comment by ujjwalpathaak** [comment]
> I think the error notice is a little too aggressive. A better option might be to automatically set the date to today if the user clicks the time picker before the date is set.

So should I raise a new pull request with these changes?

**Comment by ragesoss** [comment]
Yes, that's fine.
